projects/fal/src/fal/app.py

```diff
@@ -142,15 +142,14 @@ def _print_logs():
         try:
             with httpx.Client() as client:
                 retries = 100
-                while retries:
+                for _ in range(retries):
                     resp = client.get(info.url + "/health")
 
                     if resp.is_success:
                         break
-                    elif resp.status_code != 500:
+                    elif resp.status_code not in (500, 404):
```

Comment:
not sure why this happens but seems like sometimes we start to check for health too early - the controller did not start the app yet - so we get 404s